sympy/polys/matrices/sdm.py

```diff
@@ -851,6 +853,68 @@ def is_lower(self):
         """
         return all(i >= j for i, row in self.items() for j in row)
 
+    def lll(A, delta=QQ(3, 4)) -> 'SDM':
+        """
+        Performs the Lenstra–Lenstra–Lovász (LLL) basis reduction algorithm.
+        See [1]_ and [2]_.
+
+        Parameters
+        ==========
+
+        delta : QQ, optional
+            The Lovász parameter. Must be in the interval (0.25, 1), with larger
+            values producing a more reduced basis. The default is 0.75 for
+            historical reasons.
+
+        Returns
+        =======
+
+        The reduced basis as an SDM over ZZ.
+
+        Examples
+        ========
+
+        >>> from sympy.polys.domains import ZZ, QQ
+        >>> from sympy.polys.matrices.sdm import SDM
+        >>> x = SDM(
```

Comment:
I think it would be better not to duplicate this docstring in several places. Probably just having it once in the DomainMatrix.lll method is good. The other docstrings can just reference the DomainMatrix.lll method. All that needs to be said here for example is that this will call the dense DDM.lll routine.